docs/other-topics/hooks.md

```diff
@@ -2,404 +2,254 @@
-  beforeUpdate(instance, options)
-  beforeSave(instance, options)
-  beforeUpsert(values, options)
-
-[... creation/update/destruction happens ...]
-
-(5)
-  afterCreate(instance, options)
-  afterDestroy(instance, options)
-  afterUpdate(instance, options)
-  afterSave(instance, options)
-  afterUpsert(created, options)
-(6)
-  afterBulkCreate(instances, options)
-  afterBulkDestroy(options)
-  afterBulkUpdate(options)
-```
-
-## Declaring Hooks
-
-Arguments to hooks are passed by reference. This means, that you can change the values, and this will be reflected in the insert / update statement. A hook may contain async actions - in this case the hook function should return a promise.
-
-There are currently three ways to programmatically add hooks:
-
-```js
-// Method 1 via the .init() method
-class User extends Model {}
-User.init({
-  username: DataTypes.STRING,
-  mood: {
-    type: DataTypes.ENUM,
-    values: ['happy', 'sad', 'neutral']
-  }
-}, {
-  hooks: {
-    beforeValidate: (user, options) => {
-      user.mood = 'happy';
-    },
-    afterValidate: (user, options) => {
-      user.username = 'Toni';
-    }
-  },
-  sequelize
-});
-
-// Method 2 via the .addHook() method
-User.addHook('beforeValidate', (user, options) => {
-  user.mood = 'happy';
-});
-
-User.addHook('afterValidate', 'someCustomName', (user, options) => {
-  return Promise.reject(new Error("I'm afraid I can't let you do that!"));
-});
-
-// Method 3 via the direct method
-User.beforeCreate(async (user, options) => {
-  const hashedPassword = await hashPassword(user.password);
-  user.password = hashedPassword;
-});
-
-User.afterValidate('myHookAfter', (user, options) => {
-  user.username = 'Toni';
-});
-```
+Sequelize.hooks.addListener('beforeInit', () => {
+  console.log('A new sequelize instance is being created');
+});
+```
+
+## Instance Sequelize Hooks
+
+`Sequelize` instances support the following hooks:
+
+| Hook name                             | Async | Run when                                                   |
+|---------------------------------------|-------|------------------------------------------------------------|
+| `beforeDefine`, `afterDefine`         | ❌     | A new `Model` class is being registered                    |
+| `beforeQuery`, `afterQuery`           | ✅     | A SQL query is being run                                   |
+| `beforeBulkSync`, `afterBulkSync`     | ✅     | `sequelize.sync` is called                                 |
+| `beforeConnect`, `afterConnect`       | ✅     | Whenever a new connection to the database is being created |
+| `beforeDisconnect`, `afterDisconnect` | ✅     | Whenever a connection to the database is being closed      |
+
+:::info
+
+All [Model hooks](#model-sequelize-hooks) are also fired on the sequelize instance:
+
+```typescript
+import { Sequelize } from '@sequelize/core';
+
+const sequelize = new Sequelize(/* options */);
+
+// This will be called whenever findAll is called on any model.
```

Comment:
Commenting it here, but counts for the others as well.

I feel a bit nitpicky commenting on this, but later on we mention that findAll/findOne has been called. Should we update that everywhere?